### PR TITLE
add certificate utilities

### PR DESCRIPTION
**Pull request by yhung119**
- java bindings
- cert is constant now for add cert
- add_cert can have const args
- go bindings
- python bindings
- cpp bindings

### CODE REVIEW

File: src/ffi/cpp/crypto.cpp [hunk 1]
```diff
@@ -41,8 +41,14 @@ peacemakr::Key::Key(symmetric_cipher cipher,
 }
 
 peacemakr::Key::Key(symmetric_cipher cipher, const std::string &bytes) {
-  m_key_ =
-      peacemakr_key_new_bytes(cipher, (uint8_t *)bytes.data(), bytes.size());
+  // If the key size is small it's probably a symmetric key, otherwise it's
+  // probably a PEM file
+  if (bytes.size() <= 32) {
```
Comment (interstellarPotato) [inline src/ffi/cpp/crypto.cpp:46]: v nice

Comment (interstellarPotato) [reply src/ffi/cpp/crypto.cpp:46]: May be 32 bytes could be declared as a class constant?

File: src/core/src/Key.c [hunk 2]
```diff
@@ -291,19 +295,18 @@ peacemakr_key_t *peacemakr_key_new_from_password(
     const uint8_t *password, const size_t password_len, const uint8_t *salt,
     const size_t salt_len, const size_t iteration_count) {
   EXPECT_NOT_NULL_RET(password, "buffer is null\n")
-  EXPECT_TRUE_RET((password_len >= 0),
+  EXPECT_TRUE_RET((password_len != 0),
```
Comment (interstellarPotato) [inline src/core/src/Key.c:298]: v nice :)

**Review by interstellarPotato — COMMENTED** [review]
Thank you! A few minor comments :) 

Do we need any negative testing for this PR (i.e invalid cert)?

File: src/core/src/Key.c [hunk 3]
```diff
@@ -374,13 +377,223 @@ peacemakr_key_new_from_master(symmetric_cipher cipher,
...
+    if (rc != 1) {
+      PEACEMAKR_OPENSSL_LOG;
+      X509_REQ_free(request);
+      return false;
+    }
+  }
+
+  // Add Common Name
+  if (cn != NULL && cn_len != 0) {
+    rc =
```
Comment (yhung119) [inline src/core/src/Key.c:419]: nit: extra line break?

File: src/core/src/Key.c [hunk 4]
```diff
@@ -374,13 +377,223 @@ peacemakr_key_new_from_master(symmetric_cipher cipher,
...
+
+#define CERTIFICATE "CERTIFICATE"
+#define PUBLIC_KEY "PUBLIC KEY"
+
+#define PEACEMAKR_X509_ERR(ctx)                                                \
+  do {                                                                         \
+    int cert_error = X509_STORE_CTX_get_error((ctx));                          \
+    const char *msg = X509_verify_cert_error_string(cert_error);               \
+    PEACEMAKR_ERROR("X509 Error: %s", msg);                                    \
+  } while (0)
```
Comment (yhung119) [inline src/core/src/Key.c:522]: nit: why does this error is a while(0)?

**Review by interstellarPotato — APPROVED** [review]

**Pull request merged**